api/client/src/main/java/org/projectnessie/client/auth/oauth2/OAuth2ClientConfig.java

```diff
@@ -205,8 +205,7 @@ URI getResolvedImpersonationTokenEndpoint() {
   @Value.Lazy
   Optional<HttpAuthentication> getBasicAuthentication() {
     return getClientSecretSupplier()
-        .map(Supplier::get)
-        .map(s -> BasicAuthenticationProvider.create(getClientId(), s));
+        .map(secretSupplier -> BasicAuthenticationProvider.create(getClientId(), secretSupplier));
```

Comment:
This has an implication for the OAuth client: now each request to the token endpoint will recompute the header.

Reply:
To avoid that, maybe we can keep both `getClientSecret()` and `getClientSecretProvider()` (without deprecating the first one). Then create the basic authentication using the first non-empty optional.

Reply:
Keeping `getClientSecret()` LGTM

Reply:
It might have to wait until I'm back :-)